### PR TITLE
[RecoveryServices.Backup] Removing name param from container

### PR DESCRIPTION
**Pull request by siddharth7**
## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [X] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ContainerTests.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public ContainerTests(Xunit.Abstractions.ITestOutputHelper output)
             XunitTracingInterceptor.AddToContext(_logger);
         }
 
-        [Fact]
+        [Fact(Skip = "This workload is not supported anymore")]
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ContainerTests.cs]: Remove this test if it is not supported - same for all the other skips

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ItemTests.cs [hunk 2]
```diff
@@ -30,7 +30,7 @@ public ItemTests(Xunit.Abstractions.ITestOutputHelper output)
             XunitTracingInterceptor.AddToContext(_logger);
         }
 
-        [Fact]
+        [Fact(Skip = "This workload is not supported anymore")]
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ItemTests.cs]: See comment above - remove skipped tests if this is no longer supported

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Container/GetAzureRmRecoveryServicesBackupContainer.cs [hunk 3]
```diff
@@ -44,14 +44,6 @@ public class GetAzureRmRecoveryServicesBackupContainer : RSBackupVaultCmdletBase
         [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage")]
         public string BackupManagementType { get; set; }
 
-        /// <summary>
-        /// Friendly name of the container(s) to be fetched. This will be deprecated.
-        /// </summary>
-        [Parameter(Mandatory = false, Position = 3,
-            HelpMessage = ParamHelpMsgs.Container.Name)]
-        [ValidateNotNullOrEmpty]
-        public string Name { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Container/GetAzureRmRecoveryServicesBackupContainer.cs]: You cannot remove a parameter unless you had previously deprecated it.  Instead, you should map this to 'FriendlyName' and add a deprecation attribute.  See: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md

Comment (siddharth7) [reply src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Container/GetAzureRmRecoveryServicesBackupContainer.cs]: It has been marked depreciated for a while now, you can view the changes in AzureWorkloadProviderHelper.cs.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs [hunk 4]
```diff
@@ -194,23 +194,12 @@ public AzureWorkloadProviderHelper(ServiceClientAdapter serviceClientAdapter)
         {
             string vaultName = (string)providerData[CmdletModel.VaultParams.VaultName];
             string vaultResourceGroupName = (string)providerData[CmdletModel.VaultParams.ResourceGroupName];
-            string name = (string)providerData[CmdletModel.ContainerParams.Name];
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs]: See the file change above - remove this file 'set'

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ContainerTests.cs [hunk 5]
```diff
@@ -39,7 +39,7 @@ public void TestAzureSqlGetContainers()
                 _logger, PsBackupProviderTypes.AzureSql, "Test-AzureSqlGetContainers");
         }
 
-        [Fact]
+        [Fact(Skip = "This workload is not supported anymore")]
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ContainerTests.cs]: You need to remove these unsupported scenario tests, not skip them

**Comment by markcowl** [comment]
@siddharth7  Note please make these changes in the next few hours, or this PR will not be merged an no breaking changes will be possible.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs (hunk 4 above)
Comment (siddharth7) [reply src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs]: Done

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ContainerTests.cs (hunk 5 above)
Comment (siddharth7) [reply src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Test/ScenarioTests/AzureSql/ContainerTests.cs]: Can I take this is next release? We have a whole cleanup to do related to this workload

**Pull request merged**